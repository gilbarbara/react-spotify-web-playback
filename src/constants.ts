export const ERROR_TYPE = {
  ACCOUNT: 'account',
  AUTHENTICATION: 'authentication',
  INITIALIZATION: 'initialization',
  PLAYBACK: 'playback',
  PLAYER: 'player',
} as const;

export const STATUS = {
  ERROR: 'ERROR',
  IDLE: 'IDLE',
  INITIALIZING: 'INITIALIZING',
  READY: 'READY',
  RUNNING: 'RUNNING',
  UNSUPPORTED: 'UNSUPPORTED',
} as const;

export const TYPE = {
  DEVICE: 'device_update',
  FAVORITE: 'favorite_update',
  PLAYER: 'player_update',
  PROGRESS: 'progress_update',
  STATUS: 'status_update',
  TRACK: 'track_update',
} as const;
