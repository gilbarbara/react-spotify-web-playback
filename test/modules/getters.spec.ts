import {
  getBgColor,
  getLocale,
  getMergedStyles,
  getSpotifyLink,
  getSpotifyLinkTitle,
  getSpotifyURIType,
} from '~/modules/getters';

import { TRANSPARENT_COLOR } from '~/constants';

describe('getBgColor', () => {
  it('should return the background color', () => {
    expect(getBgColor('#f04')).toBe('#f04');
  });

  it('should return the transparent color', () => {
    expect(getBgColor('transparent')).toBe(TRANSPARENT_COLOR);
  });
});

describe('getLocale', () => {
  it('should return a merged locale', () => {
    expect(getLocale({ currentDevice: 'Selected device ' })).toMatchSnapshot();
  });
});

describe('getMergedStyles', () => {
  it('should return a merged styles', () => {
    expect(getMergedStyles({ bgColor: 'transparent', height: 100 })).toMatchSnapshot();
  });
});

describe('getSpotifyLink', () => {
  it('should return a Spotify link', () => {
    expect(getSpotifyLink('spotify:track:63DTXKZi7YdJ4tzGti1Dtr')).toBe(
      'https://open.spotify.com/track/63DTXKZi7YdJ4tzGti1Dtr',
    );
  });
});

describe('getSpotifyLinkTitle', () => {
  it('should return a formatted title', () => {
    expect(getSpotifyLinkTitle('Hommer', getLocale().title)).toBe('Hommer on SPOTIFY');
  });
});

describe('getSpotifyURIType', () => {
  it.each([
    ['spotify:album:51QBkcL7S3KYdXSSA0zM9R', 'album'],
    ['spotify:artist:7A0awCXkE1FtSU8B0qwOJQ', 'artist'],
    ['spotify:episode:6r8OOleI5xP7qCEipHvdyK', 'episode'],
    ['spotify:playlist:5kHMGRfZHORA4UrCbhYyad', 'playlist'],
    ['spotify:show:5huEzXsf133dhbh57Np2tg', 'show'],
    ['spotify:track:0gkVD2tr14wCfJhqhdE94L', 'track'],
    ['spotify:user:gilbarbara', 'user'],
    ['spotify', ''],
  ])('%p should return %p', (value, expected) => {
    expect(getSpotifyURIType(value)).toBe(expected);
  });
});
