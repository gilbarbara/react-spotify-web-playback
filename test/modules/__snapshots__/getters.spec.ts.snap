// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`getLocale > should return a merged locale 1`] = `
{
  "currentDevice": "Selected device ",
  "devices": "Devices",
  "next": "Next",
  "otherDevices": "Select other device",
  "pause": "Pause",
  "play": "Play",
  "previous": "Previous",
  "removeTrack": "Remove from your favorites",
  "saveTrack": "Save to your favorites",
  "title": "{name} on SPOTIFY",
  "volume": "Volume",
}
`;

exports[`getMergedStyles > should return a merged styles 1`] = `
{
  "activeColor": "#1cb954",
  "altColor": "#ccc",
  "bgColor": "rgba(0, 0, 0, 0)",
  "color": "#333",
  "errorColor": "#ff0026",
  "height": 100,
  "loaderColor": "#ccc",
  "loaderSize": 32,
  "sliderColor": "#666",
  "sliderHandleBorderRadius": "50%",
  "sliderHandleColor": "#000",
  "sliderHeight": 4,
  "sliderTrackBorderRadius": 4,
  "sliderTrackColor": "#ccc",
  "trackArtistColor": "#666",
  "trackNameColor": "#333",
}
`;
